lib/matplotlib/axes/_axes.py

```diff
@@ -3331,15 +3345,22 @@ def grouped_bar(self, heights, *, positions=None, group_spacing=1.5, bar_spacing
         # place the bars, but only use numerical positions, categorical tick labels
         # are handled separately below
         bar_containers = []
-        for i, (hs, label, color) in enumerate(zip(heights, labels, colors)):
-            lefts = (group_centers - 0.5 * group_distance + margin_abs
-                     + i * (bar_width + bar_spacing_abs))
+        for i, (hs, label, color, hatch_pattern) in enumerate(
+                zip(heights, labels, colors, hatches)):
+            lefts = (
+                group_centers - 0.5 * group_distance + margin_abs
+                + i * (bar_width + bar_spacing_abs)
+            )
```

Comment:
nit but can you get your autoformatter to turn off this style of formatting:
```python 
(
                group_centers - 0.5 * group_distance + margin_abs
                + i * (bar_width + bar_spacing_abs)
            )
```
The mpl codebase doesn't use this convention for parenthesis and it's hard to read. The mpl guidelines can be found at https://matplotlib.org/devdocs/devel/coding_guide.html

Reply:
Looks like you forgot these lines when you reverted all the other style changes.